### ISSUE TITLE
[Feature] Update function for the non Store Version

### ISSUE THREAD

**Issue by Dtrieb** [Hofknecht/SystemTrayMenu#185]
It would be nice, if I have an update function (or only show me if there is a new version on github).
Thanks.

**Comment by Hofknecht** [comment]
sound great
(also Peter somewhen suggested this)

accepted

**Comment by Hofknecht** [comment]
collecting ideas for this feature

e.g. steam makes it even in background and gets then installed after restart, you can vote the update and see many informations (maybe too much, but as idea)
![image](https://user-images.githubusercontent.com/52528841/121234968-9c6c7280-c894-11eb-829c-7afa2c5a051f.png)
![image](https://user-images.githubusercontent.com/52528841/121234981-a0989000-c894-11eb-8f72-6bd6f478891e.png)

**Comment by Dtrieb** [comment]
First would be enough that there is a message that a new version is on GitHub. An give a link to the release page.

In the second step we could make it like you write it's done by steam.

**Comment by Hofknecht** [comment]
yes is a good idea to start with a small info and later we can improve, Peter made something similar in his main project there we would have template
https://github.com/topeterk/HitCounterManager/blob/master/Sources/Update.cs

**Comment by Hofknecht** [comment]
![2021-06-14 18_37_18-Window](https://user-images.githubusercontent.com/52528841/121927745-b4823d00-cd3f-11eb-82c2-b6cf7bfc6e48.png)
![2021-06-14 18_37_25-Window](https://user-images.githubusercontent.com/52528841/121927747-b51ad380-cd3f-11eb-82e8-0e480585b5ed.png)
![2021-06-14 18_37_36-Window](https://user-images.githubusercontent.com/52528841/121927750-b51ad380-cd3f-11eb-8596-8ab718b0170c.png)
![2021-06-14 18_36_29-Window](https://user-images.githubusercontent.com/52528841/121927752-b51ad380-cd3f-11eb-8c31-10d0a562a3c9.png)

notepad++ also nice example

also a nice project is https://github.com/ravibpatel/AutoUpdater.NET which we can not use because we have .net core 3.1

but basically such updater download a installer/updater which executes the update

**Comment by Hofknecht** [comment]
https://github.com/ravibpatel/AutoUpdater.NET can be added now with .net 6.
would be nice to have installer / updater, to have really a benefit 
we do not wanto to slow down the start by checking for new version or maybe as option or maybe just check very rare times
or/and manually e.g. by right click context menu, check for updates
(not a big benefit without any automatic update, because you can already today right click 'Help/Faq' -> download)
or maybe just add check for updates there in context menu and go to release site showing a message if the latest one is greater than current one otherwise tell user something like 'already latest version'

**Comment by topeterk** [comment]
For the new Xamarin front end of the HitCounterManager I am currently starting a timer on the application start that will check for updates in the background, so i starts up quickly and after the online data was fetched I then notify the user that there is a new version and just print (as noted above already) some notes about the new release and a link to get it. (This only runs if the users wants to have this online check, so one hast to opt-in to get this behavior.

This might also work for STM, the icon could change with a explamation mark or a earth symbol (most commen for "internet") or anything similar, so the user doesn't get a message box out of nowhere. Or creating a balloon tip (if this still exists in Win11) when the menu gets open. Or just send a notification to the windows notification center that an update is available (as you might know when a mail arrives or windows tells you to run the next update)
